ubuntu-noble-standard: macaulay2 needs boost_regex

```
  [macaulay2-1.24.11]   [spkg-install] CMake Error at /usr/lib/x86_64-linux-gnu/cmake/Boost-1.83.0/BoostConfig.cmake:141 (find_package):
  [macaulay2-1.24.11]   [spkg-install]   Could not find a package configuration file provided by "boost_regex"
  [macaulay2-1.24.11]   [spkg-install]   (requested version 1.83.0) with any of the following names:
  [macaulay2-1.24.11]   [spkg-install] 
  [macaulay2-1.24.11]   [spkg-install]     boost_regexConfig.cmake
  [macaulay2-1.24.11]   [spkg-install]     boost_regex-config.cmake
  [macaulay2-1.24.11]   [spkg-install] 
  [macaulay2-1.24.11]   [spkg-install]   Add the installation prefix of "boost_regex" to CMAKE_PREFIX_PATH or set
  [macaulay2-1.24.11]   [spkg-install]   "boost_regex_DIR" to a directory containing one of the above files.  If
  [macaulay2-1.24.11]   [spkg-install]   "boost_regex" provides a separate development package or SDK, be sure it
  [macaulay2-1.24.11]   [spkg-install]   has been installed.
 ```
https://github.com/passagemath/passagemath/actions/runs/12229492505/job/34109544012?pr=527#step:11:3851

--- a/build/pkgs/boost_cropped/distros/debian.txt
+++ b/build/pkgs/boost_cropped/distros/debian.txt
@@ -1,1 +1,2 @@
 libboost-dev
+libboost-regex-dev

